packages/@glimmer/application/src/application.ts

```diff
@@ -70,14 +121,26 @@ export default class Application implements Owner {
   private _rendering = false;
   private _rendered = false;
   private _scheduled = false;
-  private _result: RenderResult;
+
+  protected builder: Builder;
+  protected loader: Loader;
+  protected renderer: Renderer;
 
   protected _notifiers: Notifier[] = [];
 
   constructor(options: ApplicationOptions) {
     this.rootName = options.rootName;
     this.resolver = options.resolver;
-    this.document = options.document || (typeof window !== 'undefined' && window.document);
+
+    this.document = options.document || DEFAULT_DOCUMENT;
+    this.loader = options.loader || new RuntimeLoader(this.resolver);
```

Comment:
As we talked about lets remove the defaults, add an assertion, and then we need to update the blueprints.